tests/manage/z_cluster/cluster_expansion/conftest.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+import logging
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework import config
+from ocs_ci.ocs.node import (
+    get_worker_nodes,
+    add_new_node_and_label_it,
+    add_new_node_and_label_upi,
+    taint_nodes,
+)
+from ocs_ci.ocs import machine as machine_utils
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def add_nodes():
+    """
+    Test for adding worker nodes to the cluster while IOs
+    """
+
+    def factory(ocs_nodes=False, node_count=3, taint_label=None):
+        """
+        Args:
+            ocs_nodes (bool): True if new nodes are OCS, False otherwise
+            node_count (int): Number of nodes to be added
+            taint_label (str): Taint label to be added
+
+        """
+
+        new_nodes = []
+        if config.ENV_DATA["platform"].lower() in constants.CLOUD_PLATFORMS:
+            dt = config.ENV_DATA["deployment_type"]
+            if dt == "ipi":
+                machines = machine_utils.get_machinesets()
+                log.info(
+                    f"The worker nodes number before expansion {len(get_worker_nodes())}"
+                )
+                for machine in machines:
+                    new_nodes.append(
+                        add_new_node_and_label_it(machine, mark_for_ocs_label=ocs_nodes)
+                    )
+
+                log.info(
+                    f"The worker nodes number after expansion {len(get_worker_nodes())}"
+                )
+
+            else:
+                log.info(
+                    f"The worker nodes number before expansion {len(get_worker_nodes())}"
+                )
+                if config.ENV_DATA.get("rhel_workers"):
+                    node_type = constants.RHEL_OS
+                else:
+                    node_type = constants.RHCOS
+
+                new_nodes.append(
+                    add_new_node_and_label_upi(
+                        node_type, node_count, mark_for_ocs_label=ocs_nodes
+                    )
+                )
+                log.info(
+                    f"The worker nodes number after expansion {len(get_worker_nodes())}"
+                )
+
+        elif config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM:
+            # pytest.skip(
+            #     "Skipping add node in Vmware platform due to "
+            #     "https://bugzilla.redhat.com/show_bug.cgi?id=1844521"
+            # )
+            # Issue to remove skip code https://github.com/red-hat-storage/ocs-ci/issues/2403
```

Comment:
Why do we have this skip here and commented out? Is it some leftover?

Reply:
Add nodes is skipped on VMWARE, @vavuthu can provide more details on this

Reply:
Ok, but I see that whole skip here is commented out so it doesn't take any effect.
In the old code I see there is:
```python
            pytest.skip(	
                "Skipping add node in Vmware platform due to "	
                "https://bugzilla.redhat.com/show_bug.cgi?id=1844521"	
            )
```
Before those commented lines, but it's not here so probably it's missing here?

Reply:
Corrected